gradoop-data-integration/src/test/java/org/gradoop/dataintegration/transformation/PropagatePropertyToNeighborTest.java

```diff
@@ -0,0 +1,183 @@
+/*
+ * Copyright © 2014 - 2018 Leipzig University (Database Research Group)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.gradoop.dataintegration.transformation;
+
+import org.gradoop.common.model.impl.properties.PropertyValue;
+import org.gradoop.dataintegration.transformation.impl.PropertyTransformation;
+import org.gradoop.flink.model.GradoopFlinkTestBase;
+import org.gradoop.flink.model.api.operators.UnaryGraphToGraphOperator;
+import org.gradoop.flink.model.impl.epgm.LogicalGraph;
+import org.gradoop.flink.util.FlinkAsciiGraphLoader;
+import org.junit.Test;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.List;
+
+
+/**
+ * Tests for the {@link PropagatePropertyToNeighbor} operator.
+ */
+public class PropagatePropertyToNeighborTest extends GradoopFlinkTestBase {
+
+  /**
+   * A comparator ordering property values by type first.
+   * This will effectively be able to compare any property value.
+   * Whether this order makes any sense is not relevant for this test, it is just used to make sure
+   * that two list-typed properties have the same order.
+   */
+  private static Comparator<PropertyValue> byTypeFirst = Comparator
+    .comparing((PropertyValue pv) -> pv.getType().getSimpleName())
+    .thenComparing(Comparator.naturalOrder());
+
+  /**
+   * The loader with the graphs used in this test.
+   */
+  private FlinkAsciiGraphLoader loader = getLoaderFromString(
+    "input1[" + "(s1:Source {p1: 1, p2: 1.1d})-[e1:edge1]->(t:Target {t: 0})" +
+      "(s2:Source {p1: \"\"})-[e2:edge2]->(t)" +
+      "(s1)-[e12:edge1]->(t2:Target2 {t: 0})" +
+      "(s2)-[e22:edge2]->(t2)" +
+      "] input2 [" +
+      "(v:Vertex {t: 1})-->(v)" +
+      "]");
+
+  /**
+   * Test the operator propagating a property to two vertices.
+   *
+   * @throws Exception when the execution in Flink fails.
+   */
+  @Test
+  public void testPropagateDirected() throws Exception {
+    LogicalGraph input = loader.getLogicalGraphByVariable("input1");
+    UnaryGraphToGraphOperator operator = new PropagatePropertyToNeighbor("Source", "p1", "t");
+    // We have to update the vertices manually because the ascii loader does not support lists.
+    LogicalGraph expected = input.transformVertices((v, c) -> {
+      if (!"Target".equals(v.getLabel()) && !"Target2".equals(v.getLabel())) {
```

Comment:
Why not `if (!v.getLabel().equals("Target") && !v.getLabel().equals("Target2")) {` ? I can not read this without getting a headache ^^

Reply:
Strange style i agree. Only makes sense if you consider "null" as "not equals". But since every element of the graph has a label (even default label "") by default, you should stay with the according coding style of the project.

Reply:
You did not changed it. Was this intentional or did you forget it?

Reply:
Sorry, I changed it, since you both dislike this style. (Although I don't think it is bad style, just a matter of preference.)